File: trunk/agentsystem_ros_tutorials/opencv_ros_bridge_tutorial/src/publish_screenpoint.py
#!/usr/bin/env python

# http://www.ros.org/wiki/ROS/Tutorials/WritingPublisherSubscriber%28python%29
import roslib; roslib.load_manifest('opencv_ros_bridge_tutorial')
import rospy;
import math;
from geometry_msgs.msg import PointStamped

pub = rospy.Publisher('/image_painted/screenpoint', PointStamped)
rospy.init_node('publish_screenpoint')
msg = PointStamped()
i = 0;
while not rospy.is_shutdown():
    msg.header.stamp = rospy.Time.now()
    msg.point.x = 320 + 180*math.sin(i)
    msg.point.y = 240 + 180*math.cos(i)
    i += 0.1
    print msg;
    pub.publish(msg)
    rospy.sleep(0.1)

File: opencv_ros_bridge_tutorial/src/publish_screenpoint.cpp
// http://www.ros.org/wiki/ROS/Tutorials/WritingPublisherSubscriber%28c%2B%2B%29
#include "ros/ros.h"
#include "geometry_msgs/PointStamped.h"

#include <math.h>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "publish_screenpoint");

  ros::NodeHandle n;
  ros::Publisher pub = n.advertise<geometry_msgs::PointStamped>("/image_painted/screenpoint", 1000);

  ros::Rate loop_rate(10);
  double i = 0;
  while (ros::ok())
    {
      geometry_msgs::PointStamped msg;
      msg.header.stamp = ros::Time().now();
      msg.point.x = 320 + 180*sin(i);
      msg.point.y = 240 + 180*cos(i);
      i += 0.1;
      ROS_INFO("%5.1f %5.1f", msg.point.x, msg.point.y);
      pub.publish(msg);
      ros::spinOnce();
      loop_rate.sleep();
    }

  return 0;
}
